torch/csrc/jit/codegen/cuda/expr_simplifier.cpp

```diff
@@ -280,6 +280,45 @@ Val* recurseDown(Val* value, std::function<Val*(Val*)> rule) {
 
 } // namespace
 
+namespace reg {
+
+enum class RegisterType { GeneralPurposed, Uniform, Immediate, Unknown };
+
+RegisterType getRegisterType(Val* value, const VarInfo& var_info) {
```

Comment:
This seems to be a promotion rule for register type, is that right?

Reply:
Yes, correct